main/forum/editthread.php

```diff
@@ -0,0 +1,220 @@
+}
+
+/* Resource Linker */
+
+if (isset($_POST['add_resources']) AND $_POST['add_resources'] == get_lang('Resources')) {
+    $_SESSION['formelements']	= $_POST;
+    $_SESSION['origin']			= $_SERVER['REQUEST_URI'];
+    $_SESSION['breadcrumbs']	= $interbreadcrumb;
+    header('Location: ../resourcelinker/resourcelinker.php');
+    exit;
+}
+
+/* Header */
+
+$htmlHeadXtra[] = <<<JS
+    <script>
+    $(document).on('ready', function() {
+        $('#reply-add-attachment').on('click', function(e) {
+            e.preventDefault();
+
+            var newInputFile = $('<input>', {
+                type: 'file',
+                name: 'user_upload[]'
+            });
+
+            $('[name="user_upload[]"]').parent().append(newInputFile);
+        });
+    });
+    </script>
+JS;
+
+if ($origin == 'learnpath') {
+    Display::display_reduced_header();
+} else {
+    Display :: display_header(null);
+}
+
+handle_forum_and_forumcategories();
+
+// Action links
+echo '<div class="actions">';
+echo '<span style="float:right;">'.search_link().'</span>';
+echo '<a href="viewforum.php?forum='.Security::remove_XSS($_GET['forum']).'&'.api_get_cidreq().'">'.
+    Display::return_icon('back.png',get_lang('BackToForum'),'',ICON_SIZE_MEDIUM).'</a>';
+echo '</div>';
+
+$threadData = getThreadInfo($threadId, $cId);
+
+// Set forum attachment data into $_SESSION
+getAttachedFiles($current_forum['forum_id'], 0, 0);
+$values = show_update_thread_form(
+    $current_forum,
+    $forum_setting,
+    $threadData
+);
+
+if (!empty($values) && isset($values['SubmitPost'])) {
+    // update thread in table forum_thread.
+    update_thread($values);
+    
+    $option_chek = isset($values['thread_qualify_gradebook']) ? $values['thread_qualify_gradebook'] : null; // values 1 or 0
+    if (1 == $option_chek) {
+        $id = $values['thread_id'];
+        $title_gradebook = Security::remove_XSS(stripslashes($values['calification_notebook_title']));
+        $value_calification = $values['numeric_calification'];
+        $weight_calification = $values['weight_calification'];
+        $description = '';
+        $session_id = api_get_session_id();
+
+        $link_info = GradebookUtils::is_resource_in_course_gradebook(
+            api_get_course_id(),
+            5,
+            $id,
+            $session_id
+        );
+        $link_id = $link_info['id'];
+
+        if (!$link_info) {
+            GradebookUtils::add_resource_to_course_gradebook(
+                $values['category_id'],
+                api_get_course_id(),
+                5,
+                $id,
+                $title_gradebook,
+                $weight_calification,
+                $value_calification,
+                $description,
+                1,
+                api_get_session_id()
+            );
+        } else {
+            Database::query('UPDATE '.$table_link.' SET weight='.$weight_calification.' WHERE id='.$link_id.'');
```

Comment:
Make sure $weight_calification (which should be name $weight_qualification, really) is not empty. If empty, set it to "0".
If $link_id is empty, don't execute the query.
Use Doctrine.